taiga_contrib_ldap_auth/services.py

```diff
@@ -62,17 +67,29 @@ def ldap_login_func(request):
     # (or any other attribute)
     login_input = request.DATA.get('username', None)
     pass_input = request.DATA.get('password', None)
-
-    # TODO: sanitize before passing to LDAP server?
-    username, email, full_name = connector.login(login = login_input,
-                                                 password = pass_input)
-
-    user = register_or_update(username = username,
-                              email = email,
-                              full_name = full_name)
-
-    data = make_auth_response_data(user)
-    return data
+    try:
+        # TODO: make sure these fields are sanitized before passing to LDAP server!
+        username, email, full_name = connector.login(login = login_input, password = password_input)
+    except connector.LDAPUserLoginError as ldap_error:
```

Comment:
@martijnbastiaan I took a look - not two, but many. Local database, LDAP, seemingly SMB and PAM and whatever.

[Here](https://github.com/nextcloud/server/blob/51688986df4a4d3d99ede3576c1639917a3da738/lib/private/User/Manager.php#L131): it seems that when a user is looked up upon login attempt, all enabled auth backends are queried (sequentially?). Upon match, a user object is returned, and the user will have its backend implementation set (as a property in the user object? I'm illiterate when it comes to PHP).

Backends [implement an interface](https://github.com/nextcloud/server/blob/b4f36d44c43aac0efdc6c70ff8e46473341a9bfe/lib/private/User/Backend.php) for common actions, such as password verification, password changing (or can perhaps have a "password reset" link), even creating and removing users.

Anyway, I don't think we get this fancy here, which is why the change of `loginFormType` in `taiga-front` config. Potentially, there could be a `universal` login form type, but I haven't looked at Taiga close enough to say if there's already a similar interface abstraction, a-la [façade](https://en.wikipedia.org/wiki/Facade_pattern). Or if there's already a plugin for that.